refinery/file_store/models.py

```diff
@@ -54,34 +54,11 @@ def _mkdir(path):
     return True
 
 
-# configure and create file store directories
-if not settings.FILE_STORE_DIR:
-    settings.FILE_STORE_DIR = 'file_store'  # relative to MEDIA_ROOT
-
-# absolute path to the file store root dir
-FILE_STORE_BASE_DIR = os.path.join(settings.MEDIA_ROOT,
-                                   settings.FILE_STORE_DIR)
-# create this directory in case it doesn't exist
-if not os.path.isdir(FILE_STORE_BASE_DIR):
-    _mkdir(FILE_STORE_BASE_DIR)
-
-# temp dir should be located on the same file system as the base dir
-FILE_STORE_TEMP_DIR = os.path.join(FILE_STORE_BASE_DIR, 'temp')
-# create this directory in case it doesn't exist
-if not os.path.isdir(FILE_STORE_TEMP_DIR):
-    _mkdir(FILE_STORE_TEMP_DIR)
-
-# To make sure we can move uploaded files into file store quickly instead of
-# copying
-if not settings.FILE_UPLOAD_TEMP_DIR:
-    settings.FILE_UPLOAD_TEMP_DIR = FILE_STORE_TEMP_DIR
-# To keep uploaded files always on disk
-if not settings.FILE_UPLOAD_MAX_MEMORY_SIZE:
-    settings.FILE_UPLOAD_MAX_MEMORY_SIZE = 0
-
-# http://stackoverflow.com/q/4832626
-FILE_STORE_BASE_URL = \
-    urlparse.urljoin(settings.MEDIA_URL, settings.FILE_STORE_DIR) + '/'
+# create data storage directories
+if not os.path.isdir(settings.FILE_STORE_BASE_DIR):
+    _mkdir(settings.FILE_STORE_BASE_DIR)
+if not os.path.isdir(settings.FILE_STORE_TEMP_DIR):
+    _mkdir(settings.FILE_STORE_TEMP_DIR)
```

Comment:
If there were a problem creating a directory, `_mkdir` just returns `False` instead of throwing the exception... I'm not sure that that behavior is useful here? Is there a reason not to use `mkdirs`?

Reply:
Return value of `_mkdir()` is never checked so I'll remove corresponding statements. `os.makedirs()` is problematic because it [raises an exception if the leaf directory already exists](https://docs.python.org/2/library/os.html#os.makedirs) which is often the case.